docs(ipfs): review api/cli section MTA-3493 (#1968)
scaleway/docs-content: labs/ipfs/api-cli/ipfs-cli.mdx

  - pinning
  - ipfs-pinning
---
Scaleway IPFS Pinning allows you to permanently store a copy of your data from the public IPFS network on a Scaleway-owned node, thus providing you with an added layer of performance in the region of your choice. That way, your data remains available and accessible, even when your local machine is offline.

With your content now available on both your local node and our resilient and reliable external nodes, you drastically minimize the risk of disruptions thanks to redundancy. You can also enhance your data storage efficiency by pinning your content to your Scaleway node, and then remove it from your local node.

> Supported by [Protocol Labs](https://protocol.ai/), [Kubo](https://docs.ipfs.tech/install/command-line/) is the official implementation of IPFS. Written in Go programming language, it is available on Windows, Linux and Mac. This page shows you how you can [create and manage your volumes](/labs/ipfs/api-cli/volumes-operations) and [pins](/labs/ipfs/api-cli/pin-operations) using the Kubo CLI.

Reviewer: ```suggestion
Supported by [Protocol Labs](https://protocol.ai/), [Kubo](https://docs.ipfs.tech/install/command-line/) is the official implementation of IPFS. Written in Go programming language, it is available on Windows, Linux, and Mac. This page shows you how you can [create and manage your volumes](/labs/ipfs/api-cli/volumes-operations) and [pins](/labs/ipfs/api-cli/pin-operations) using the Kubo CLI.
```

## Requirements to use our service through the Kubo client

<Macro id="iam-requirements" />

<Message type="requirement">
    - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/)
    - You have generated your [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
    - You have downloaded and installed the [IPFS Desktop](/labs/ipfs/reference-content/install-ipfs-desktop/) or the [Kubo client](https://docs.ipfs.tech/install/command-line/)

>     - You have create a [volume](/labs/ipfs/api-cli/volumes-operations/)

Reviewer: ```suggestion
    - You have created a [volume](/labs/ipfs/api-cli/volumes-operations/)
```

</Message>

## Initializing and starting your Kubo client

1. Enter the following CLI command to initialize your local IPFS node:

    ```
      ipfs init
    ```

2. Enter the following CLI command to launch the IPFS background process, effectively turning your device into a functioning node in the IPFS network:

    ```
    ipfs daemon
    ```

    <Message type="tip">
        If you wish to use the same terminal for future commands, you need to run `ipfs daemon &`. This will run the daemon in the background and return control to the terminal.
    </Message>

## Adding Scaleway as a remote pinning service

> Enter the following command to create add Scaleway as a remote service in your local Kubo via its CLI. Make sure to replace the example values with your own, using the configuration table below:

Reviewer 2: ```suggestion
Enter the following command to add Scaleway as a remote service in your local Kubo via its CLI. Make sure to replace the example values with your own, using the configuration table below:
```

```
  ipfs pin remote service add scaleway https://<REGION_ID>.ipfs.labs.scw.cloud/<VOLUME_ID> <SECRET_KEY>
```

## Pinning a content using its CID

This operation shows you how you can pin a content using its CID through your own Kubo client.

Run the following CLI command to pin a content using its CID: 

```
ipfs pin remote add --service=scaleway <CID>
```

> You should get an output similar to this, providing details about the content: 

Reviewer 2: ```suggestion
You should get an output similar to te following, providing details about the content: 
```

Run the following CLI command to remove a pin from your content using its CID:

```
ipfs pin remote rm --service=scaleway --cid=<CID>
```

## Configuration table

| Parameter    | Description                                                                                                                                                                                                                               |
| :----------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `REGION_ID`  | Create volumes and pins in this given region. Possible values are `fr-par`, `pl-waw` and `nl-ams`. Default value is `fr-par`.                                                                                                             |
| `VOLUME_ID`  | ID of the volume you wish to pin your content to. You can find it using either the [Scaleway console](https://console.scaleway.com/ipfs/volumes) or the [Scaleway API/CLI](/labs/ipfs/api-cli/volumes-operations)                         |

> | `SECRET_KEY` | The secret part of your [API key](/identity-and-access-management/iam/concepts/#api-key)                                                                                                                                                  |

Reviewer 2: ```suggestion
| `SECRET_KEY` | Your [API secret key](/identity-and-access-management/iam/concepts/#api-key)                                                                                                                                                  |
```
